pkg/blob/azure_msi.go

```diff
@@ -0,0 +1,96 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blob
+
+// This code has been copy-pasted from https://github.com/google/go-cloud/blob/master/blob/azureblob/azureblob.go
```

Comment:
Nit: dot at the end of the second comment.

Reply:
Do we track an issue for this?
I guess we can't upgrade beyond 0.19.0 because of #2798 ?

Reply:
Actually we might want this solution to stay here for a little longer, because the interface of gocould.dev isn't really handy.

There are two ways to open a bucket: via `OpenBucket` and via the URL opener. `OpenBucket` doesn't include the MSI authentication flow. `OpenBucketURL` does include MSI, but it doesn't take `accountName` as an argument, rather fetches it from the environment, so we would need something like

```
os.setenv("AZURE_ACCOUNT_NAME", accountName)
return blob.OpenBucketURL("azure://" + containerName)
```

which doesn't look very nice

Reply:
So it's not copy pasted but adapted